machine/v1alpha1/types_openstack.go

```diff
@@ -366,3 +372,62 @@ type RootVolume struct {
 	// Deprecated: deviceType will be silently ignored. There is no replacement.
 	DeprecatedDeviceType string `json:"deviceType,omitempty"`
 }
+
+// blockDeviceStorage is the storage type of a block device to create and
+// contains additional storage options.
+type BlockDeviceStorage struct {
+	// type is the type of block device to create.
+	// This can be either "Volume" or "Local".
+	// +kubebuilder:validation:Enum="Volume";"Local"
+	// +kubebuilder:validation:Required
+	Type BlockDeviceType `json:"type"`
+
+	// volume contains additional storage options for a volume block device.
+	// +union
+	// +optional
+	Volume *BlockDeviceVolume `json:"volume,omitempty"`
+}
+
+// blockDeviceVolume contains additional storage options for a volume block device.
+type BlockDeviceVolume struct {
+	// type is the volume type of the volume.
+	// If omitted, the default volume type will be used.
+	// +optional
+	Type string `json:"type,omitempty"`
+
+	// availabilityZone is the volume availability zone to create the volume in.
+	// If omitted, the availability zone of the server will be used.
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+}
+
+// additionalBlockDevice is a block device to attach to the server.
+// +union
+type AdditionalBlockDevice struct {
+	// name of the block device in the context of a machine.
+	// If the block device is a volume, the Cinder volume will be named
+	// as a combination of the machine name and this name.
+	// Also, this name will be used for tagging the block device.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// size is the size in GiB of the block device.
+	// +kubebuilder:validation:Required
+	Size int `json:"size"`
+
+	// storage specifies the storage type of the block device and
+	// additional storage options.
```

Comment:
Godoc should include explanation of the valid values for the enum

Reply:
`BlockDeviceStorage.Type` has Godoc with valid values for the enum.
However for `AdditionalBlockDevice.Storage`, I'm not sure what enum we have?

Reply:
I think I was looking at this upside down, IIUC, the structure is
```
additionalBlockDevices:
- name: foo
  sizeGiB: 1234
  storage:
    type: Volume
    volume:
      type: my-osp-specific-type
      availabilityZone: somewhere
```
I think you can ignore the comment